util/make_timeline.py

```diff
@@ -97,14 +139,29 @@ def parse_file(args):
 
     entries = []
     started = False
-
+    encounter_sets = []
     with args.file as file:
+        # If searching for encounters, divert and find start/end first3
+        if args.search_fights:
+            encounter_sets = find_fights_in_file(file)
+            # If all we want to do is list encounters, stop here and give to the user.
+            if args.search_fights < 0:
+                return [f'{i + 1}. {e_info[0]} {e_info[1]} {e_info[2]}' for i, e_info in enumerate(encounter_sets)]
+            elif args.search_fights > len(encounter_sets):
+                raise Exception('Selected fight index not in selected ACT log.')
+        # Scan the file until the start timestamp
         for line in file:
-            # Scan the file until the start timestamp
-            if not started and line[14:26] != args.start:
+            start_time = end_time = 0
+            if args.search_fights:
+                start_time = (encounter_sets[args.search_fights - 1][0])
+                end_time = (encounter_sets[args.search_fights - 1][1])
+            else:
+                start_time = args.start
+                end_time = args.end
+            if not started and start_time not in (line[14:26], line[14:29]):
                 continue
 
-            if line[14:26] == args.end:
+            if end_time in (line[14:26], line[14:29]):
```

Comment:
I'm not sure I understand what's going on with the 14:26 and 14:29, could you explain it?

Reply:
That's an artifact from before the elided microsecond issue was fixed. Definitely unnecessary now.